packages/edit-site/src/components/sidebar-dataviews/default-views.js

```diff
@@ -68,21 +69,59 @@ const DEFAULT_POST_BASE = {
 	layout: defaultLayouts[ LAYOUT_LIST ].layout,
 };
 
+function useDataViewItemCounts( { postType } ) {
+	const { records, totalItems } = useEntityRecords( 'postType', postType, {
+		per_page: -1,
```

Comment:
I probably followed the [same rabbit hole](https://github.com/WordPress/gutenberg/pull/62028#discussion_r1616054976) as @Souptik2001 here. 

Unfortunately I couldn't find another way either. 

`getEntityRecordsTotalItems` won't return anything unless there's been a `getEntityRecords` query to populate the state, so we have to fetch the items.

Ideally, I guess, there'd be an options request with a limited database query `SELECT COUNT` to set the `X-WP-Total` header or something... I'm not sure